src/models/transaction.ts

```diff
@@ -0,0 +1,350 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { CREATE_TOKEN_TX_VERSION, DEFAULT_TX_VERSION, DECIMAL_PLACES, TOKEN_INFO_VERSION, TX_WEIGHT_CONSTANTS, MAX_INPUTS, MAX_OUTPUTS } from '../constants';
+import { crypto, encoding, util } from 'bitcore-lib';
+import helpers from '../utils/helpers';
+import Input from './input';
+import Output from './output';
+import { CreateTokenTxInvalid, MaximumNumberInputsError, MaximumNumberOutputsError } from '../errors';
+import buffer from 'buffer';
+
+type optionsType = {
+  version?: number,
+  weight?: number,
+  nonce?: number,
+  timestamp?: number | null,
+  parents?: string[],
+  tokens?: string[],
+  hash?: string | null,
+  name?: string | null,
+  symbol?: string | null,
+};
+
+const defaultOptions: optionsType = {
+  version: DEFAULT_TX_VERSION,
+  weight: 0,
+  nonce: 0,
+  timestamp: null,
+  parents: [],
+  tokens: [],
+  hash: null,
+  name: null,
+  symbol: null,
+}
+
+class Transaction {
+  inputs: Input[];
+  outputs: Output[];
+  version: number;
+  weight: number;
+  nonce: number;
+  timestamp: number | null;
+  parents: string[];
+  tokens: string[];
+  hash: string | null;
+  name: string | null;
+  symbol: string | null;
+
+  constructor(inputs: Input[], outputs: Output[], options: optionsType = defaultOptions) {
+    const newOptions = Object.assign(defaultOptions, options);
+    const { version, weight, nonce, timestamp, parents, tokens, hash, name, symbol } = newOptions;
+
+    this.inputs = inputs;
+    this.outputs = outputs;
+    this.version = version;
+    this.weight = weight;
+    this.nonce = nonce;
+    this.timestamp = timestamp;
+    this.parents = parents;
+    this.tokens = tokens;
+    this.hash = hash;
+    this.name = name;
+    this.symbol = symbol;
+  }
+
+  /**
+   * Returns a string with the short version of the tx hash
+   * Returns {first12Chars}...{last12Chars}
+   *
+   * @return {string}
+   * @memberof Transaction
+   * @inner
+   *
+   */
+  getShortHash(): string {
+    return this.hash === null ? '' : `${this.hash.substring(0,12)}...${this.hash.substring(52,64)}`;
+  }
+
+  /**
+   * Return transaction data to sign in inputs
+   * 
+   * @return {Buffer}
+   * @memberof Transaction
+   * @inner
+   */
+  getDataToSign(): Buffer {
```

Comment:
Should we cache it? I mean, the common use is to sign/verify the transaction, and all inputs will get the same data.

Reply:
Makes sense! Done